extensions/notification/configManager/MongoConfigManager.js

```
@@ -0,0 +1,312 @@
+                    method: 'MongoConfigManager._setMetastoreChangeStream',
+                });
+                return cb(error);
+            }
+            return cb();
+        });
+    }
+
+    /**
+     * Get bucket notification configuration
+     *
+     * @param {String} bucket - bucket
+     * @return {Object|undefined} - configuration if available or undefined
+     */
+    async getConfig(bucket) {
+        const startTime = Date.now();
+        // return cached config for bucket if it exists
+        const cachedConfig = this._cachedConfigs.get(bucket);
+        if (cachedConfig) {
+            const delay = (Date.now() - startTime) / 1000;
+            onConfigManagerConfigGet(true, delay);
+            return {
+                bucket,
+                notificationConfiguration: cachedConfig,
+            };
+        }
+        try {
+            // retreiving bucket metadata from the metastore
+            const bucketMetadata = await this._metastore.findOne({ _id: bucket });
+            const notificationConfiguration = bucketMetadata?.value?.notificationConfiguration;
+            const delay = (Date.now() - startTime) / 1000;
+            onConfigManagerConfigGet(false, delay);
+            if (!notificationConfiguration) {
+                return undefined;
+            }
+            // caching the bucket configuration
+            this._cachedConfigs.add(bucket, notificationConfiguration);
+            onConfigManagerCacheUpdate('add');
+            return {
+                bucket,
+                notificationConfiguration
+            };
+        } catch (err) {
+            this._logger.error('An error occured when getting notification ' +
+                'configuration of bucket', {
+                method: 'MongoConfigManager.getConfig',
+                bucket,
+                error: err.message,
+            });
+            return undefined;
```

should we return the err here?